server.rb

```diff
@@ -21,33 +21,33 @@
 
     pkey = OpenSSL::PKey::RSA.new(public_key)
 
-    if pkey.verify(OpenSSL::Digest::SHA1.new, Base64.decode64(signature), payload)
-      logger.info("verification succeeded")
-      payload = JSON.parse(payload)
+    unless pkey.verify(OpenSSL::Digest::SHA1.new, Base64.decode64(signature), payload)
+      logger.info("verification failed")
+      halt 412
```

Comment:
412確かヘッダーの`If-`系が失敗したときに返すステータスコードなので、それ以外で返すのはよくないかも

>  The 412 (Precondition Failed) status code indicates that one or more
>   conditions given in the request header fields evaluated to false when
>   tested on the server.
https://tools.ietf.org/html/rfc7232#section-4.2

```
 3. Precondition Header Fields .....................................13
      3.1. If-Match ..................................................13
      3.2. If-None-Match .............................................14
      3.3. If-Modified-Since .........................................16
      3.4. If-Unmodified-Since .......................................17
      3.5. If-Range ..................................................18
```
https://tools.ietf.org/html/rfc7232